src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex_Create.cs

```diff
@@ -381,5 +394,167 @@ private static void Free(bool ignoreCase, HashSet<string> identifiers, HashSet<s
+                // For the first call to 'TryProcessNextContainer', offset is just after the ':' token.
+                // We parse the string from left to right appending characters to the given stringBuilder
+                // until we reach a '.' character which indicates we have reached the end of documentation comment id
+                // for the next outermost containing symbol.
+                // For this example, we will popupate 'N' in the stringBuilder and 'newOffset' will be pointing to 'C'.
+
+                if (stringBuilder.Length != 0)
+                {
+                    // If this is not the first call, then we append the '.' char 
+                    // as a separator for the next containing symbol that we are going to process.
+                    // For the above example, if we processing offset 'C', the stringBuilder will
+                    // have 'N' from last invocation, so we append '.' and then process the rest of the string.
+                    Debug.Assert(originalDocCommentId[offset - 1] == '.');
+                    stringBuilder.Append('.');
+                }
+
+                newOffset = offset;
+                while (newOffset < originalDocCommentId.Length)
+                {
+                    var ch = originalDocCommentId[newOffset++];
+                    switch (ch)
+                    {
+                        case '.':
+                            // We have found the separator char indicating the next containing symbol.
+                            return true;
+
+                        case '`':
+                            // '`' is used to specific arity of a generic type in symbol's documentation commend id.
+                            // We need to include this arity for a complete documentation commend id representation of the symbol.
+                            stringBuilder.Append(ch);
+                            continue;
+
+                        case '#':
+                        case '(':
+                        case '[':
+                            // These tokens indicate that we are now processing a member symbol such as
+                            // a constructor, method argument list, indexer argument list, etc.
+                            // We cannot have any more container symbols in the documentation commend id.
+                            return false;
+
+                        default:
+                            // Continue processing and appending characters as long as we see valid identifier part characters.
+                            // A non-identifier character indicates we do not have any more container symbols in the documentation commend id.
+                            if (!syntaxFacts.IsIdentifierPartCharacter(ch))
+                            {
+                                return false;
+                            }
+
+                            stringBuilder.Append(ch);
+                            continue;
```

Comment:
when we need to do the FindRefs search, can we not just use a regex search for this sort of thing?  having anotehr custom parser seems unfortunate.

Reply:
yes, switched to regex now.